modules/skybrowser/src/screenspaceskytarget.cpp

```diff
@@ -0,0 +1,430 @@
+        _shader->setUniform(_uniformCache.modelTransform, modelTransform);
+        _shader->setUniform(_uniformCache.lineColor, color);
+        _shader->setUniform(
+            _uniformCache.viewProj,
+            global::renderEngine->scene()->camera()->viewProjectionMatrix()
+        );
+
+        glBindVertexArray(rendering::helper::vertexObjects.square.vao);
+        glDrawArrays(GL_TRIANGLES, 0, 6);
+
+        glEnable(GL_CULL_FACE);
+
+        _shader->deactivate();
+    }
+
+    void ScreenSpaceSkyTarget::update() {
+        if (_isLocked) {
+            glm::dvec3 localCamera = skybrowser::equatorialToLocalCamera(
+                _lockedCoordinates
+            );
+
+            if (_useRadiusAzimuthElevation) {
+                // Keep the set radius
+                glm::vec3 position = _raePosition.value().x * glm::vec3(localCamera);
+                _raePosition = cartesianToRae(position);
+            }
+            else {
+                _cartesianPosition = skybrowser::localCameraToScreenSpace3d(
+                    localCamera
+                );
+            }
+           
+        }
+    }
+
+    void ScreenSpaceSkyTarget::setDimensions(glm::vec2 dimensions) {
+        // To avoid flooring of the size of the target, multiply by factor of 100
+        // Object size is really the pixel size so this calculation is not exact
+        _objectSize = glm::ivec2(dimensions * 100.f);
+    }
+
+    glm::dvec3 ScreenSpaceSkyTarget::directionGalactic() const {
+        glm::vec3 localCamera = _cartesianPosition.value();
+
+        if (_useRadiusAzimuthElevation) {
+            localCamera = raeToCartesian(_raePosition.value());
+        }
+
+        glm::dmat4 rotation = glm::inverse(
+            global::navigationHandler->camera()->viewRotationMatrix()
+        );
+        glm::dvec4 position = glm::dvec4(localCamera, 1.0);
+
+        return glm::normalize(rotation * position);
+    }
+
+    // Update the scale of the target (the height of the target in relation to the 
+    // OpenSpace window)
+    void ScreenSpaceSkyTarget::setScaleFromVfov(float verticalFov) {
+        _verticalFov = verticalFov;
+        glm::dvec2 fovs = skybrowser::fovWindow();
+        
+        // Cap the scale at small scales so it is still visible
+        float heightRatio = verticalFov / fovs.y;
+        float smallestHeightRatio = _showRectangleThreshold.value() / fovs.y;
+        
+        _scale = std::max(heightRatio, smallestHeightRatio);
+    }
+
+    void ScreenSpaceSkyTarget::setFovFromScale() {
+        glm::dvec2 fovs = skybrowser::fovWindow();
+        _verticalFov = _scale * fovs.y;
+    }
+
+    bool ScreenSpaceSkyTarget::isLocked() const {
+        return _isLocked;
+    }
+
+    bool ScreenSpaceSkyTarget::isAnimated() {
+        return _isAnimated;
+    }
+
+    void ScreenSpaceSkyTarget::startAnimation(glm::dvec3 equatorialCoordsEnd, 
+                                              bool shouldLockAfter) {
```

Comment:
this starting bracket should move to the next line, since the signature is more than one line